plugin/js/showpass-custom.js

```diff
@@ -277,14 +277,30 @@
 					const client_id = gaCookie.substring(6); // using the example above, this will return "1194072907.1685136322"
 					const session_id = client_id.split(".")[1]; // ["1194072907", "1685136322"]
 
-					if (!isNaN(Number(client_id)) && !isNaN(Number(session_id))) {
+					if (
+						!isNaN(Number(client_id)) &&
+						!isNaN(Number(session_id))
+					) {
 						let url = new URL(iFrame.src);
-						url.searchParams.append('client_id', client_id);
-						url.searchParams.append('session_id', session_id);
+						url.searchParams.append("parent_client_id", client_id);
+						url.searchParams.append(
+							"parent_session_id",
+							session_id
+						);
 						iFrame.src = url.toString();
 					}
 				}
 			}
+
+			// Pass the parent page's referrer to our iFrame.
+			// When the referrer is unavailable (ie. direct visit), the web-app
+			// should not inject and GA4 defaults to the default behaviour.
+			const referrer = document.referrer || "";
+			if (referrer) {
```

Comment:
Do you also need to check if `referrer !== ""`
?

Reply:
Just checking, since we do that check for the cookie up above.

Reply:
Hmm cant remember exactly why we did that for the cookie tbh lol but I _think_ theres a reason for cookies..